### PR TITLE
CIS Azure: Implement 7.2, 7.3

### PR DESCRIPTION
**Pull request by orestisfl**
### Summary of your changes
Ensure Virtual Machines are utilizing Managed Disks

I can't test failure E2E because when creating a VM without a managed disk, the option is disabled:
> Availability zone requires managed disks.

### Related Issues
- Related: https://github.com/elastic/cloudbeat/issues/1258

### Checklist

- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?

- [x] Generate rule metadata using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rules-metadata)
- [x] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

> **Note** Make sure to bump cloudbeat [`version/policy.go`](https://github.com/elastic/cloudbeat/blob/main/version/policy.go#L20) after merging this PR and drafting a new release.

### CODE REVIEW

File: bundle/compliance/cis_azure/rules/cis_7_4/test.rego [hunk 1]
```diff
@@ -3,10 +3,10 @@ package compliance.cis_azure.rules.cis_7_4
 import data.compliance.policy.azure.data_adapter
 import data.lib.test
 
-generate_encryption_settings(type) = {
+generate_encryption_settings(type) = {"encryption": {
```
Comment (orestisfl) [inline bundle/compliance/cis_azure/rules/cis_7_4/test.rego:6]: Unrelated to 7.2, just wanted to make sure 7.4 handles missing "encryption" data

File: bundle/compliance/cis_azure/rules/cis_7_2/rule.rego [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+		common.calculate_result(has_managed_disk),
+		{"Resource": data_adapter.resource},
+	)
+}
+
+default has_managed_disk := false
+
+has_managed_disk {
+	data_adapter.properties.storageProfile.osDisk.managedDisk.id != ""
+}
```
Comment (oren-zohar) [inline bundle/compliance/cis_azure/rules/cis_7_2/rule.rego:21]:
```suggestion
has_managed_disk {
	data_adapter.properties.storageProfile.osDisk.managedDisk.id != ""
} else = false
```

File: bundle/compliance/cis_azure/rules/cis_7_2/rule.rego [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	result := common.generate_result_without_expected(
+		common.calculate_result(has_managed_disk),
+		{"Resource": data_adapter.resource},
+	)
+}
+
+default has_managed_disk := false
+
+has_managed_disk {
+	data_adapter.properties.storageProfile.osDisk.managedDisk.id != ""
```
Comment (oren-zohar) [inline bundle/compliance/cis_azure/rules/cis_7_2/rule.rego:20]: is it an empty string? or is it completely missing?

File: bundle/compliance/cis_azure/rules/cis_7_2/test.rego [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+import data.compliance.policy.azure.data_adapter
+import data.lib.test
+
+valid_managed_disk = {
+	"id": "/subscriptions/sub-id/resourceGroups/cloudbeat-resource-group-1695893762/providers/Microsoft.Compute/disks/cloudbeatVM_OsDisk_1_e736df07f12142a9a2784ea8de9084ce",
+	"resourceGroup": "cloudbeat-resource-group-1695893762",
+	"storageAccountType": "Standard_LRS",
+}
+
+generate_vm(managed_disk) = {
```
Comment (oren-zohar) [inline bundle/compliance/cis_azure/rules/cis_7_2/test.rego:12]: should be under test_data

File: bundle/compliance/cis_azure/rules/cis_7_2/test.rego [hunk 5]
```diff
@@ -0,0 +1,121 @@
+package compliance.cis_azure.rules.cis_7_2
+
+import data.compliance.policy.azure.data_adapter
+import data.lib.test
+
+valid_managed_disk = {
```
Comment (oren-zohar) [inline bundle/compliance/cis_azure/rules/cis_7_2/test.rego:6]: should be under test_data

File: bundle/compliance/cis_azure/rules/cis_7_3/rule.rego [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import data.compliance.policy.azure.data_adapter
+import data.compliance.policy.azure.disk.ensure_encryption as audit
+
+finding = result {
+	# filter
+	data_adapter.is_attached_disk
+
+	# set result
+	result := common.generate_result_without_expected(
+		common.calculate_result(audit.is_encryption_enabled),
```
Comment (oren-zohar) [inline bundle/compliance/cis_azure/rules/cis_7_3/rule.rego:13]: I can't find the audit function

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: bundle/compliance/cis_azure/rules/cis_7_2/rule.rego (hunk 3 above)
Comment (orestisfl) [reply bundle/compliance/cis_azure/rules/cis_7_2/rule.rego:20]: I am just checking the id exists as a sanity check, it's the one field that should always be there.

File: bundle/compliance/cis_azure/rules/cis_7_2/test.rego (hunk 4 above)
Comment (orestisfl) [reply bundle/compliance/cis_azure/rules/cis_7_2/test.rego:12]: Should I really move it if it's only used in this file?

File: bundle/compliance/cis_azure/rules/cis_7_2/test.rego (hunk 5 above)
Comment (orestisfl) [reply bundle/compliance/cis_azure/rules/cis_7_2/test.rego:6]: Should I really move it if it's only used in this file?

File: bundle/compliance/cis_azure/rules/cis_7_3/rule.rego (hunk 6 above)
Comment (orestisfl) [reply bundle/compliance/cis_azure/rules/cis_7_3/rule.rego:13]: It's an import:

```
import data.compliance.policy.azure.disk.ensure_encryption as audit
```

Comment (orestisfl) [reply bundle/compliance/cis_azure/rules/cis_7_3/rule.rego:13]: Which is shared with rule 7.4 (already committed)

File: bundle/compliance/cis_azure/rules/cis_7_2/test.rego (hunk 5 above)
Comment (oren-zohar) [reply bundle/compliance/cis_azure/rules/cis_7_2/test.rego:6]: Yes this is mostly for consistency and readability, if we'll get reusability of it in the future for free is just a plus

**Review by oren-zohar — COMMENTED** [review]
Approving since the main comments were addressed. I would still prefer to move the parts I pointed out the test_data file but your call

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**